### PR TITLE
Misc. updates to amp-analytics.md

### PR DESCRIPTION
**Pull request by avimehta**
- Updated details of click selector
- *Max conditions in visibilitySpec
- Added details of sampleSpec

### CODE REVIEW

File: extensions/amp-analytics/amp-analytics.md [hunk 1]
```diff
@@ -336,12 +336,37 @@ The `triggers` attribute describes when an analytics request should be sent. It
...
+
+  As an example, following configuration can be used to sample 50% of the requests based on random input or at 1% based on client id.
+  ```
+  'triggers': {
+    'sampledOnRandom': {
+      'on': 'visible',
+      'request': 'request',
+      'sampleSpec': {
+        'sampleOn': '${random}',
+        'threshold': 50,
```
Comment (rudygalfi) [inline extensions/amp-analytics/amp-analytics.md]: should this be 0.5?

File: extensions/amp-analytics/amp-analytics.md [hunk 2]
```diff
@@ -336,12 +336,37 @@ The `triggers` attribute describes when an analytics request should be sent. It
   - `selector` (required when `on` is set to `click`) This configuration is used on conjunction with the `click` trigger. Please see below for details.
   - `scrollSpec` (required when `on` is set to `scroll`) This configuration is used on conjunction with the `scroll` trigger. Please see below for details.
   - `timerSpec` (required when `on` is set to `timer`) This configuration is used on conjunction with the `timer` trigger. Please see below for details.
+  - `sampleSpec` This object is used to define how the requests can be sampled before they are sent. This setting allows sampling based on random input or other platform supported vars. The object contains configuration to specify an input that is used to generate a hash and a threshold that the hash must meet.
+    - `sampleOn` This string template is expanded by filling in the platform variables and then hashed to generate a number to sample based on.
```
Comment (rudygalfi) [inline extensions/amp-analytics/amp-analytics.md]: "... then hashed to generate a number for the purposes of the sampling logic described under `threshold` below."

File: extensions/amp-analytics/amp-analytics.md [hunk 3]
```diff
@@ -336,12 +336,37 @@ The `triggers` attribute describes when an analytics request should be sent. It
...
+    - `threshold` This configuration is used to filter out requests that do not meet the criteria. For a request to go through to the analytics vendor, the following logic should be true `HASH(sampleOn) < threshold`.
```
Comment (rudygalfi) [inline extensions/amp-analytics/amp-analytics.md]: "do not meet particular criteria: For a request ..."

File: extensions/amp-analytics/amp-analytics.md [hunk 4]
```diff
@@ -336,12 +336,37 @@ The `triggers` attribute describes when an analytics request should be sent. It
...
+      'sampleSpec': {
+        'sampleOn': '${clientId(cookieName)}',
+        'threshold': 1,
+      },
+    },
+  },
+  ```
 
-#### Page visible trigger (`"on": "visible"`)
+#### Page visible trigger (`"on": "visible"`)
```
Comment (rudygalfi) [inline extensions/amp-analytics/amp-analytics.md]: extra space

**Comment by rudygalfi** [comment]
LGTM after addressing a few comments

File: extensions/amp-analytics/amp-analytics.md (hunk 1 above)
Comment (avimehta) [inline extensions/amp-analytics/amp-analytics.md]: nope. This is in percentage.

**Comment by avimehta** [comment]
@pbakaus This PR contains the changes related to `sampleSpec` that I was talking about.

**Pull request merged**